test_simple_graph.py

```diff
@@ -0,0 +1,77 @@
+import pytest
+from simple_graph import Graph
+@pytest.fixture
+def populated():
+    g = Graph()
+    g.add_node(5)
+    g.add_node(7)
+    g.add_node("Hello")
+    g.add_edge(5, 7)
+    g.add_edge(5, "Hello")
+    return g
+
+
+def test_graph_constructor():
+    assert Graph().graph == {}
+
+
+def test_nodes(populated):
+    for val in [5, 7, "Hello"]:
+        assert val in populated.graph
+    assert 27 not in populated.graph
+
+
+def test_edges(populated):
+    assert populated.edges() == [[5, 7], [5, "Hello"]]
+
+
+def test_add_node():
+    g = Graph()
+    g.add_node(70)
+    assert 70 in g.graph
+
+
+def test_add_mutable(populated):
+    with pytest.raises(TypeError):
+        populated.add_node([1, 3])
+
+
+def test_add_edge(populated):
+    populated.add_edge(7, 5)
+    assert [7, 5] in populated.edges()
+
+
+def test_del_node(populated):
+    populated.del_node(5)
+    assert 5 not in populated.graph
```

Comment:
You need more complex tests here.  you haven't got near enough tests to cover possible edge cases for deleting a node, which is why the problems I note above aren't breaking your tests.